python/ray/autoscaler/updater.py

```diff
@@ -172,14 +186,21 @@ def ssh_cmd(self, cmd, connect_timeout=120, redirect=None, verbose=False):
                 "NodeUpdater: running {} on {}...".format(
                     pretty_cmd(cmd), self.ssh_ip),
                 file=self.stdout)
-        force_interactive = "set -i && source ~/.bashrc && "
+        # TODO(hartikainen): ssh command used to begin with 'set -i', but
+        # was removed since bash >= 4.4 evaluates it incorrectly. Figure out a
+        # better way to handle this.
+        # For more, see:
+        # - https://unix.stackexchange.com/q/364617
+        # - https://github.com/ray-project/ray/pull/2061#discussion_r188838635
```

Comment:
Yeah this is kind of unfortunate, especially since sourcing bashrc doesn't usually work unless you `set -i`. Is it possible to attempt set -i, and fall back if that fails?

Reply:
I think that should be possible. I'll figure that out after I've verified that the gcp provider works correctly.

Reply:
@ericl I added a fallback logic (see commit https://github.com/ray-project/ray/pull/2061/commits/23a0066c5254449e49746bd5e43b94b66f32bfb4) to the `ssh_cmd`. Not very clean, but should work.

What prevents us from adding the path variables into, for example, `~/.profile`, which is sourced even on non-interactive shells, instead of `~/.bashrc`? Is there something in general in `~/.bashrc` that we want to be able to source? My understanding of different shells is pretty non-existent, but it seems to me that there should be a better way to handle this.